cognee/tasks/sentiment_analysis/sentiment_analysis.py

```
@@ -0,0 +1,81 @@
+from textblob import TextBlob
+from cognee.shared.logging_utils import get_logger
+
+logger = get_logger(__name__)
+
+def run_sentiment_analysis(
+    prev_question: str,
+    prev_answer: str,
+    current_question: str,
+    user_id: str
+) -> dict[str, str | float]:
+    """
+    Analyze sentiment from combined conversation context.
+
+    Args:
+        prev_question (str): Previous question text
+        prev_answer (str): Previous answer text
+        current_question (str): Current question text
+        user_id (str): ID of the user
+
+    Returns:
+        dict[str, str | float]: {
+            "user_id": str,
+            "sentiment": "positive" | "neutral" | "negative",
+            "score": float,
+            "context": str
+        }
+    """
+
+    # Validate input
+    if not any([prev_question, prev_answer, current_question]):
+        logger.warning(f"Empty input for sentiment analysis, user_id={user_id}")
```

Let's raise an exception when mandatory data is not provided. Developers that use this task need to get the feedback when they are not using it the right way. Would be nice to create a custom exception that extends `CogneeValidationError`.